command/root.go

```diff
@@ -351,3 +352,37 @@ func determineEditor(cmd *cobra.Command) (string, error) {
 
 	return editorCommand, nil
 }
+
+func ExpandAlias(args []string) ([]string, error) {
+	// TODO i don't love that context is being init'd twice. My reading of the code is that it is
+	// idempotent and okay but I don't know if that's ok to depend on.
+	if len(args) == 1 {
+		// the command is lacking a subcommand
+		return []string{}, nil
+	}
+
+	ctx := initContext()
+	cfg, err := ctx.Config()
+	if err != nil {
+		return []string{}, err
+	}
+	aliases, err := cfg.Aliases()
+	if err != nil {
+		return []string{}, err
+	}
+
+	if aliases.Exists(args[1]) {
+		expansion := aliases.Get(args[1])
+		if strings.Contains(expansion, "$") {
+			for i, a := range args[2:] {
+				expansion = strings.Replace(expansion, fmt.Sprintf("$%d", i+1), a, 1)
+			}
+			if strings.Contains(expansion, "$") {
```

Comment:
This disallows `$` appearing anywhere in alias expansion unless it's used for `$<N>` positional argument syntax. I feel that this is overly restrictive, since there may be valid uses for `$` (for example, an alias may contain a GraphQL query passed to `gh api`).

I vote that we only raise this error if there are still un-expanded `$<N>` expressions in the expansion, but that we allow `$` followed by non-number characters.